scripts/dev-restart.sh

```
@@ -0,0 +1,172 @@
+    echo "🔍 Checking port $port..."
+    
+    # Cari PID yang menggunakan port
+    local pid=$(lsof -ti:$port)
+    
+    if [ ! -z "$pid" ]; then
+        echo "⚠️  Killing process on port $port (PID: $pid)"
+        kill -9 $pid 2>/dev/null
+        sleep 1
+        
+        # Verifikasi port sudah kosong
+        local check_pid=$(lsof -ti:$port)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)